### ISSUE TITLE
updating config file in pull_asset

### ISSUE THREAD

**Issue by nsheff** [databio/refgenconf#77]
raised by @johanneskoester

Right now there's no easy way to update a config file after a pull_asset request in the conf object. in the CLI, we're doing this: 

https://github.com/databio/refgenie/blob/4b9cb4d877d516e63e0799953ad4b75406d3ca95/refgenie/refgenie.py#L670-L680

There are about 10 lines of code there that it seems like you'd want to do every time you call `pull_asset`... I can't imaging wanting to pull an asset but not update the config. We should at least allow an `update` option to `pull_asset` that does this for you, as noticed here:

https://github.com/snakemake/snakemake-wrappers/pull/30/files#diff-f0fed67ee444cc827d70a8c9ed42d06fR19

**Comment by michalstolarczyk** [comment]
I've added an `update` arg to `pull_asset` method on `dev` branch. This way you can update the `RefGenConf` object with the newly pulled asset data.

@johanneskoester, once we release the next version of `refgenconf`, `bio/refgenie/wrapper.py` could read as follows:

```python
rgc = refgenconf.RefGenConf(conf_path, writable=True)
gat, archive_data, server_url = rgc.pull_asset(genome, asset, tag, force=False, update=True)
```
and if you want the updates to be saved to the config on disk:
```python
rgc.write()
```

**Comment by nsheff** [comment]
to completely solve this as we discussed we should make sure to bump the version requirements for the new yacman functionality

**Comment by michalstolarczyk** [comment]
We've just released `refgenconf v0.6.0`, which includes `pull_asset` method updates. 

Now, just by executing: `rgc.pull_asset(genome, asset, tag)` the asset is retrieved, config file locked, updated with the asset metadata and unlocked. No writability mode manipulation is required anymore.
